motors/tacho_motor_class.c

```diff
@@ -721,16 +861,17 @@ static ssize_t speed_sp_store(struct device *dev, struct device_attribute *attr,
 			      const char *buf, size_t size)
 {
 	struct tacho_motor_device *tm = to_tacho_motor(dev);
-	int err, speed;
-
-	if (!(BIT(TM_SPEED_REGULATION_ON) & tm->ops->get_speed_regulations(tm->context)))
-		return -EOPNOTSUPP;
+	int err, speed, max_speed;
 
 	err = kstrtoint(buf, 10, &speed);
 	if (err < 0)
 		return err;
 
-	if (speed > 2000 || speed < -2000)
+	err = tm->ops->get_max_speed(tm->context, &max_speed);
+	if (err < 0)
+		return err;
+
+	if (abs(speed) > max_speed)
```

Comment:
I was thinking about this a bit more. If you make `max_speed` the no-load speed at 9V, 100% duty cycle, then realistically, the motor is never going to run faster than that. In that case, leaving the check would be OK. People will just have to be aware that speed regulation won't work near `max_speed`. This is where the `overloaded` state flag that I mentioned in a previous comment comes in.